### PR TITLE
Add org.squidowl.halloy

### PR DESCRIPTION
**Pull request by tarkah**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

---

Replaces #4292

https://github.com/squidowl/halloy

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.squidowl.halloy

File: org.squidowl.halloy.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "runtime-version": "22.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "halloy",
+    "finish-args": [
+        "--share=ipc",
+        "--filesystem=xdg-config",
+        "--filesystem=xdg-data",
```
Comment (hfiguiere) [inline org.squidowl.halloy.json:13]: the linter doesn't like these
```
{
    "errors": [
        "finish-args-arbitrary-xdg-config-access",
        "finish-args-arbitrary-xdg-data-access"
    ]
}
```

Comment (tarkah) [reply org.squidowl.halloy.json]: f72bb90 I've restricted access to our specific app folder

**Comment by hfiguiere** [comment]
bot, build org.squidowl.halloy

**Comment by tarkah** [comment]
What's the resolution to fix this? I do want access to those host xdg paths. Without those args, I can't access the halloy directory from the respective host directories. They are valid sandbox permissions so I'm not sure why this is failing as "unnecessary".

```
    "errors": [
        "finish-args-unnecessary-xdg-config-access",
        "finish-args-unnecessary-xdg-data-access"
    ]
}
```

**Comment by tarkah** [comment]
I've gone head and removed the xdg filesystem permissions and we've decided to stick w/ the XDG paths that flatpak remaps vs using the original HOST paths.

[8116740](https://github.com/flathub/flathub/pull/4293/commits/811674013dd20c6d7dc890de101a7e52bd34629a)

**Comment by hfiguiere** [comment]
bot, build org.squidowl.halloy

**Comment by hfiguiere** [comment]
you can trigger builds yourself as well.

**Comment by tarkah** [comment]
> you can trigger builds yourself as well.

Ahh good to know, thanks

**Comment by tarkah** [comment]
@barthalion I noticed the `/merge` command didn't close this, however the package did get published to flathub and the repo created.

Have I been given maintainership rights to https://github.com/flathub/org.squidowl/halloy ?

**Comment by tarkah** [comment]
@barthalion bump on getting this closed / inviting me as collaborator

**Comment by TingPing** [comment]
I invited you, however it appears some other post-creation steps failed so I'll leave this open so bart can ensure everything is as it should be.

**Comment by barthalion** [comment]
I added missing branch protection settings.

**Pull request closed without merging**